cpp/ql/src/semmle/code/cpp/ir/implementation/raw/Instruction.qll

```diff
@@ -981,14 +981,22 @@ class InheritanceConversionInstruction extends UnaryInstruction {
  * to the address of a direct non-virtual base class.
  */
 class ConvertToBaseInstruction extends InheritanceConversionInstruction {
```

Comment:
@dbartol How come `InheritanceConversionInstruction` isn't a subclass of `ConvertInstruction`? I think the name suggests that it should be, and an inheritance conversion is also a `Conversion` in the AST.

Reply:
`ConvertInstruction` is specifically for the `Convert` opcode. I'd be happy with a base class (maybe `ConversionInstruction`) that encompassed both `ConvertInstruction` and `InheritanceConversionInstruction`. Related to that, I've wondered if we should have more specific conversion opcodes, like `ConvertFloatToInteger`, `SignExtend`, `ConvertIntegerToPointer`, rather than just `Convert`.